src/Orleans.Runtime/GrainDirectory/LocalGrainDirectory.cs

```diff
@@ -569,26 +613,17 @@ public Task UnregisterAfterNonexistingActivation(ActivationAddress addr, SiloAdd
 
         public async Task UnregisterAsync(ActivationAddress address, UnregistrationCause cause, int hopCount)
         {
+            var callerMembershipVersion = default(MembershipVersion);
+
             (hopCount > 0 ? UnregistrationsRemoteReceived : unregistrationsIssued).Increment();
 
             if (hopCount == 0)
                 InvalidateCacheEntry(address);
 
             // see if the owner is somewhere else (returns null if we are owner)
-            var forwardAddress = this.CheckIfShouldForward(address.Grain, hopCount, "UnregisterAsync");
+            var forwardAddress = await this.GetPartitionOwner(address.Grain, hopCount, callerMembershipVersion, "UnregisterAsync");
 
-            // on all silos other than first, we insert a retry delay and recheck owner before forwarding
-            if (hopCount > 0 && forwardAddress != null)
-            {
-                await this.RefreshMembership();
-                forwardAddress = this.CheckIfShouldForward(address.Grain, hopCount, "UnregisterAsync");
-                if (forwardAddress != null)
-                {
-                    this.log.LogWarning($"UnregisterAsync - It seems we are not the owner of activation {address}, trying to forward it to {forwardAddress} (hopCount={hopCount})");
-                }
-            }
-
-            if (forwardAddress == null)
```

Comment:
This behavior is now consolidated into `GetPartitionOwner`, which replaces `CheckIfShouldForward`